--- a/payload.hpp
+++ b/payload.hpp
@@ -81,7 +81,7 @@ struct XdpScreencastPortal {
     XdpScreencastFlags cast_flags = XdpScreencastFlags::XDP_SCREENCAST_FLAG_NONE;
     XdpCursorMode cursor_mode = get_current_session_type() == SessionType::Wayland ? 
                                 XDP_CURSOR_MODE_EMBEDDED :
-                                XDP_CURSOR_MODE_HIDDEN;
+                                XDP_CURSOR_MODE_EMBEDDED; // prefer double cursor over no cursor
     
     // hyprland cursor mode workaround.
     // as hyprland does not support XDP_CURSOR_MODE_HIDDEN, we simply use XDP_CURSOR_MODE_EMBEDDED for it
